components/net/http_cache.rs

```diff
@@ -762,9 +762,9 @@ impl HttpCache {
         done_chan: &mut DoneChannel,
     ) -> Option<Response> {
         assert_eq!(response.status.map(|s| s.0), Some(StatusCode::NOT_MODIFIED));
-        let entry_key = CacheKey::new(&request);
+        let entry_key = CacheKey::new(request);
         if let Some(cached_resources) = self.entries.get_mut(&entry_key) {
-            for cached_resource in cached_resources.iter_mut() {
+            if let Some(cached_resource) = cached_resources.iter_mut().next() {
```

Comment:
This used to iterate all of the cached resources, but this version only iterates the first.

Reply:
The warning in this case was that this for only ran once, because it ends with a return, so there may be some logic missing if the intention was to iterate all `cached_resources`.

Reply:
Thanks. I missed this in my review.